src/apple/xcode-manager.ts

```diff
@@ -302,103 +331,971 @@ export class XcodeProject {
-    })[0];
+    return path.join(absoluteGroupPath, fileRefPath);
+  }
 
-    if (buildPhaseKey === undefined) {
+  /**
+   * Resolves the absolute path of a file reference relative to the built products directory.
+   *
+   * @param buildFile - The file reference to resolve the path of
+   * @returns The absolute path of the file reference, or undefined if the file reference is not found or has no path
+   */
+  private resolveAbsoluteFilePathRelativeToBuiltProductsDir(
+    buildFile: XcodeProjectObjectWithId<PBXFileReference>,
+  ): string | undefined {
+    debug(
+      `Resolving absolute file path relative to built products directory for file reference: ${
+        buildFile.id
+      } with path: ${buildFile.obj.path ?? ''}`,
+    );
+    const builtProductsDir = this.getBuildProductsDirectoryPath();
+    if (!builtProductsDir) {
+      debug(`Failed to resolve built products directory path`);
       return undefined;
     }
 
-    const buildPhase = this.objects.PBXSourcesBuildPhase?.[
-      buildPhaseKey.value
-    ] as PBXSourcesBuildPhase;
-    const buildPhaseFiles = buildPhase?.files ?? [];
+    return path.join(builtProductsDir, buildFile.obj.path.replace(/"/g, ''));
+  }
 
-    const baseDir = path.dirname(path.dirname(this.projectPath));
+  /**
+   * Resolves the absolute path of a file reference relative to the source root.
+   *
+   * The source root is the directory where the `.xcodeproj` file is located.
+   *
+   * @param buildFile - The file reference to resolve the path of
+   * @returns The absolute path of the file reference, or undefined if the file reference is not found or has no path
+   */
+  private resolveAbsoluteFilePathRelativeToSourceRoot(
+    buildFile: XcodeProjectObjectWithId<PBXFileReference>,
+  ): string | undefined {
+    return path.join(this.baseDir, buildFile.obj.path.replace(/"/g, ''));
+  }
 
-    return buildPhaseFiles
-      .map((file) => {
-        const fileRef = (
-          this.objects.PBXBuildFile?.[file.value] as PBXBuildFile
-        )?.fileRef;
-        if (!fileRef) {
-          return '';
-        }
-        const buildFile = fileDictionary[fileRef];
-        if (!buildFile) {
-          return '';
-        }
-        return path.join(baseDir, buildFile);
-      })
-      .filter((f: string) => f.length > 0);
+  /**
+   * Resolves the absolute path of a file reference relative to the SDK root.
+   *
+   * @param buildFile - The file reference to resolve the path of
+   * @returns The absolute path of the file reference, or undefined if the file reference is not found or has no path
+   */
+  private resolveAbsoluteFilePathRelativeToSdkRoot(
+    buildFile: XcodeProjectObjectWithId<PBXFileReference>,
+  ): string | undefined {
+    debug(
+      `Resolving absolute file path relative to SDK root for file reference: ${
+        buildFile.id
+      } with path: ${buildFile.obj.path ?? ''}`,
+    );
+
+    const sdkRoot = MacOSSystemHelpers.findSDKRootDirectoryPath();
```

Comment:
`m`: We might need to inject the `MacOSSystemHelpers` as a dependency so we can run the unit tests on Ubuntu.

Reply:
I'll explore this, because due to the nature of JavaScript we can also mock/spy patch them

Reply:
For now it can just be mocked in tests, no need to make it replaceable.